src/plugins/slack-now-playing/main.ts

```diff
@@ -0,0 +1,807 @@
+    // Prepare the status update payload
+    const statusUpdatePayload = {
+      profile: JSON.stringify({
+        status_text: statusText,
+        status_emoji: statusEmoji,
+        status_expiration: expirationTime,
+      }),
+    };
+
+    // Update the status
+    // The client now handles API errors internally
+    await client.post('users.profile.set', statusUpdatePayload);
+
+    // Update state with the new status and emoji
+    state.lastStatus = statusText;
+    state.lastEmoji = statusEmoji;
+
+  } catch (error: unknown) {
+    // Handle SlackError specifically
+    if (error instanceof SlackError) {
+      console.error(`Slack API error updating status: ${error.message}`, {
+        endpoint: error.endpoint,
+        statusCode: error.statusCode,
+        responseError: error.responseData?.error,
+      });
+    }
+    // Handle other errors
+    else if (error instanceof Error) {
+      console.error(`Error updating Slack status: ${error.message}`, {
+        name: error.name,
+        stack: error.stack
+      });
+    } else {
+      console.error(`Error updating Slack status: ${String(error)}`);
+    }
+
+    // Re-throw the error to be handled by the caller
+    throw error;
+  }
+}
+
+async function getStatusEmoji(songInfo: SongInfo, config: SlackNowPlayingConfig): Promise<string> {
+  if (songInfo.imageSrc && await uploadEmojiToSlack(songInfo, config)) {
+    return `:${config.emojiName}:`;
+  }
+
+  const randomIndex = Math.floor(Math.random() * defaultEmojis.length);
+  return defaultEmojis[randomIndex];
+}
+
+/**
+ * Uploads album art to Slack as a custom emoji
+ * @param songInfo Information about the current song
+ * @param config Plugin configuration
+ * @returns True if the emoji was successfully uploaded, false otherwise
+ */
+async function uploadEmojiToSlack(songInfo: SongInfo, config: SlackNowPlayingConfig): Promise<boolean> {
+  try {
+    // Validate configuration
+    const validationResult = validateConfig(config);
+    if (!validationResult.valid) {
+      console.error(`Cannot upload emoji to Slack: ${validationResult.errors.join(', ')}`);
+      return false;
+    }
+
+    const client = new SlackApiClient(config.token, config.cookieToken);
+
+    // Save album art to a temporary file
+    const filePath = await saveAlbumArtToFile(songInfo);
+    if (!filePath) {
+      console.warn('Failed to save album art to file');
+      return false;
+    }
+
+    // Make sure the emoji doesn't already exist
+    const emojiDeleted = await ensureEmojiDoesNotExist(config);
+    if (!emojiDeleted) {
+      console.warn('Failed to ensure emoji does not exist');
+      return false;
+    }
+
+    // Prepare the form data for the API request using formdata-node
+    const formData = new FormData();
+    formData.append('mode', 'data');
+    formData.append('name', config.emojiName);
+    try {
+      // Use fileFromPath to get a proper File object for formdata-node
+      const imageFile = await fileFromPath(filePath);
```

Comment:
here

Reply:
ok, this ain't a built-in, but you don't need an entire library for this
you can write 3 lines of code